skrub/_dataframe/_common.py

```diff
@@ -1187,3 +1208,15 @@ def with_columns(df, **new_cols):
     cols = {col_name: col(df, col_name) for col_name in column_names(df)}
     cols.update({n: make_column_like(df, c, n) for n, c in new_cols.items()})
     return make_dataframe_like(df, cols)
+
+@dispatch
+def drop(obj, col):
```

Comment:
I don’t know if `drop` is necessary, you could directly use skrub selectors: 
`df = s.select(df, ~s.cols(col))`